File: commands/message.js
'use strict';

// eslint-disable-next-line no-unused-vars
exports.Run = async function Run(caller, command, guild, lang) {
  if (!command.params.length) {
    caller.utils.createMessage(command.msg.channel.id, {
      embed: {
        color: caller.color.blue,
        title: lang.titles.use,
        fields: [
          {
            name:
              command.prefix + command.command + lang.commands.message.params,
            value: lang.commands.message.help,
          },
          {
            name: lang.example,
            value: `${command.prefix + command.command} ${
              command.msg.id
            }\n${command.prefix +
              command.command} Epic message here :sunglasses:\n\n[${
              lang.guidePage
            }](https://zira.pw/guide/${command.command})`,
          },
        ],
      },
    });
    return;
  }
  if (!guild.currentChannel) {
    caller.utils.createMessage(command.msg.channel.id, {
      embed: {
        color: caller.color.yellow,
        title: lang.titles.error,
        description: lang.errors.setChannel,
      },
    });
    return;
  }
  if (command.params.length === 1 && !isNaN(command.params[0])) {
    let message;
    try {
      message = await caller.bot.getMessage(
        guild.currentChannel,
        caller.utils.parseID(command.params[0]),
      );
    } catch (e) {
      caller.logger.warn(
        `[Message Get] ${e.code} ${e.message.replace(/\n\s/g, '')}`,
      );
      switch (e.code) {
        case 50001:
          caller.utils.createMessage(command.msg.channel.id, {
            embed: {
              color: caller.color.yellow,
              title: lang.titles.error,
              description: lang.errors.missingPermissionChannelRead,
            },
          });
          break;
        case 404:
        case 10008:
          caller.utils.createMessage(command.msg.channel.id, {
            embed: {
              color: caller.color.yellow,
              title: lang.titles.error,
              description: lang.errors.unknownMessageID,
            },
          });
          break;
        default:
          caller.utils.createMessage(command.msg.channel.id, {
            embed: {
              color: caller.color.yellow,
              title: lang.titles.error,
              description: lang.errors.generic,
            },
          });
      }
      return;
    }
    if (guild.messages.indexOf(message.id) === -1) {
      guild.messages.push(message.id);
    }
    guild.currentMessage = message.id;
    await caller.utils.updateGuild(guild);
    caller.utils.createMessage(command.msg.channel.id, {
      embed: {
        color: caller.color.green,
        title: lang.titles.complete,
        description: lang.commands.message.setID,
      },
    });
  } else {
    let message;
    try {
      message = await caller.bot.createMessage(
        guild.currentChannel,
        command.params.join(' '),
      );
    } catch (e) {
      caller.logger.warn(
        `[Message Send] ${e.code} ${e.message.replace(/\n\s/g, '')}`,
      );
      switch (e.code) {
        case 50001: // read
          caller.utils.createMessage(command.msg.channel.id, {
            embed: {
              color: caller.color.yellow,
              title: lang.titles.error,
              description: lang.errors.missingPermissionChannelRead,
            },
          });
          break;
        case 50013: // send
          caller.utils.createMessage(command.msg.channel.id, {
            embed: {
              color: caller.color.yellow,
              title: lang.titles.error,
              description: lang.errors.missingPermissionChannelSend,
            },
          });
          break;
        default:
          caller.utils.createMessage(command.msg.channel.id, {
            embed: {
              color: caller.color.yellow,
              title: lang.titles.error,
              description: lang.errors.generic,
            },
          });
      }
      return;
    }
    guild.messages.push(message.id);
    guild.currentMessage = message.id;
    await caller.utils.updateGuild(guild);
    caller.utils.createMessage(command.msg.channel.id, {
      embed: {
        color: caller.color.green,
        title: lang.titles.complete,
        description: lang.commands.message.setCreate,
      },
    });
  }
};

exports.Settings = {
  category: 0,
  command: 'message',
  show: true,
  permissions: ['manageRoles'],
  dm: false,
};

File: src/main/java/pw/zira/bot/commands/Message.java
package pw.zira.bot.commands;

import net.dv8tion.jda.core.EmbedBuilder;
import net.dv8tion.jda.core.Permission;
import net.dv8tion.jda.core.entities.MessageEmbed;
import net.dv8tion.jda.core.events.message.MessageReceivedEvent;
import net.dv8tion.jda.core.exceptions.ErrorResponseException;
import net.dv8tion.jda.core.exceptions.InsufficientPermissionException;
import pw.zira.bot.main.Zira;
import pw.zira.bot.utils.Guild;

import java.util.List;


public class Message extends Command {
    public Message(Zira zira) {
        super(zira, "message", 0, false, new Permission[]{Permission.MANAGE_CHANNEL});
    }

    @Override
    public void onCommand(MessageReceivedEvent event, List<String> params, Guild guild) {
        if (params.size() == 0) {
            String prefix = zira.settings.getPrefix();
            sendMessage(event, new EmbedBuilder()
                    .setTitle(zira.i18n.getTitle(guild.getLanguage(), "use"))
                    .setColor(zira.utils.blue)
                    .addField(String.format("%s%s %s", prefix, name, language.get("params")),
                            (String) language.get("help"), false)
                    .addField(zira.i18n.getBase(guild.getLanguage(), "example"),
                            String.format("%s%s %s\n%s%s A cool message here\n\n[%s](https://docs.zira.pw/%s)", prefix, name, event.getMessage().getId(), prefix, name, zira.i18n.getBase(guild.getLanguage(), "guidePage"), name), false)
                    .build());
            return;
        }
        if (guild.getChannel().equals(0L)) {
            sendMessage(event, new EmbedBuilder()
                    .setTitle(zira.i18n.getTitle(guild.getLanguage(), "error"))
                    .setColor(zira.utils.lightYellow)
                    .setDescription(zira.i18n.getError(guild.getLanguage(), "setChannel")).build());
            return;
        }
        boolean HasID = true;
        long id = 0;
        try {
            id = Long.parseLong(params.get(0));
        } catch (NumberFormatException e) {
            HasID = false;
        }
        MessageEmbed embed = new EmbedBuilder().setTitle(zira.i18n.getTitle(guild.getLanguage(), "complete")).setDescription((String) language.get("setID")).setColor(zira.utils.lightGreen).build();
        if (HasID) {
            try {
                net.dv8tion.jda.core.entities.Message message = event.getJDA().getTextChannelById(guild.getChannel()).getMessageById(id).complete();
                zira.db.updateGuild(guild.setMessage(message.getIdLong()));
                sendMessage(event, embed);
            } catch (ErrorResponseException e) {
                System.out.println("[Message:Get] " + e.getErrorCode() + ": " + e.getMeaning());
                switch (e.getErrorCode()) {
                    case 50001:
                        sendMessage(event, new EmbedBuilder().setTitle(zira.i18n.getTitle(guild.getLanguage(), "error")).setDescription(zira.i18n.getError(guild.getLanguage(), "missingPermissionChannelRead")).setColor(zira.utils.lightYellow).build());
                        break;
                    case 404:
                    case 10008:
                        sendMessage(event, new EmbedBuilder().setTitle(zira.i18n.getTitle(guild.getLanguage(), "error")).setDescription(zira.i18n.getError(guild.getLanguage(), "unknownMessageID")).setColor(zira.utils.lightYellow).build());
                        break;
                    default:
                        sendMessage(event, new EmbedBuilder().setTitle(zira.i18n.getTitle(guild.getLanguage(), "error")).setDescription(zira.i18n.getError(guild.getLanguage(), "generic")).setColor(zira.utils.lightYellow).build());
                }
            }
        } else {
            try {
                event.getJDA().getTextChannelById(guild.getChannel()).sendMessage(String.join(" ", params)).queue(message -> {
                    zira.db.updateGuild(guild.setMessage(message.getIdLong()));
                    sendMessage(event, embed);
                });
            } catch (InsufficientPermissionException e) {
                System.out.println("[Message:Send] " + e.getMessage());
                sendMessage(event, new EmbedBuilder().setTitle(zira.i18n.getTitle(guild.getLanguage(), "error")).setDescription("ADD STRING TO LANG\n\nMissing permission: " + e.getPermission().getName()).setColor(zira.utils.lightYellow).build());
            }
        }
    }
}
